docs/static/configuration.asciidoc

```diff
@@ -392,6 +392,9 @@ output {
 }
 ----------------------------------
 
+But be careful, the local environment of sprintf differs from the local environment of the date filter. As a result, this can create inconsistencies in the rollover of indices and other unexpected behaviours.
+This is why it is best to specify a canonical time zone ID to be used for date analysis which will avoid any inconsistencies during your use such as http://joda-time.sourceforge.net/apidocs/org/joda/time/format/DateTimeFormat.html[joda time format] or https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/format/DateTimeFormatter.html#patterns[java time format] (recommanded).
+
```

Comment:
> the local environment of sprintf differs from the local environment of the date filter

Let's talk about the "inconsistencies" you are experiencing.

 - The `@timestamp` holds a `LogStash::Timestamp` object, which by definition an instant on the UTC timeline, so formatting it with sprintf may not yield the same calendar day or clock hour as your machine-local timezone.
 - The Date filter _can_ be configured with a [`timezone`](https://www.elastic.co/guide/en/logstash/current/plugins-filters-date.html#plugins-filters-date-timezone), which it uses as plugin-local context to disambiguate strings that do not contain offset information. Whether or not this is supplied, the resulting timestamp object will be a `LogStash::Timestamp` and therefore an instant on the UTC timeline. The original offset information (as provided by the string or assumed with local context) is not retained.

Reply:
Yes, your commit seems to make things more clear, I think it's a great addition to the documentation.

Reply:
Thanks for sharing your experience and helping us clarify the docs, @SirBouBou.